typescript-rewrite/samlang-core/src/interpreter/__tests__/expression-interpreter.test.ts

```diff
@@ -0,0 +1,796 @@
+    fieldOrder: 0,
+    alias: 'f',
+    range: exampleRange,
+  };
+  const objectPattern: ObjectPattern = {
+    range: exampleRange,
+    type: 'ObjectPattern',
+    destructedNames: [objectDestructedNames, objectDestructedNamesNoAlias],
+  };
+  const objectPatternFail: ObjectPattern = {
+    range: exampleRange,
+    type: 'ObjectPattern',
+    destructedNames: [objectDestructedNamesFail],
+  };
+  const objectExpression: SamlangExpression = EXPRESSION_OBJECT_CONSTRUCTOR({
+    range: exampleRange,
+    type: intType,
+    fieldDeclarations: [
+      { range: exampleRange, type: intType, name: 'field', expression: intLiteralExpression },
+    ],
+  });
+  const objectStatement: SamlangValStatement = {
+    pattern: objectPattern,
+    range: exampleRange,
+    typeAnnotation: intType,
+    assignedExpression: objectExpression,
+  };
+  const objectStatementFail: SamlangValStatement = {
+    pattern: objectPatternFail,
+    range: exampleRange,
+    typeAnnotation: intType,
+    assignedExpression: objectExpression,
+  };
+  const variableLocalValues = new Map<string, Value>();
+  variableLocalValues.set('var', { type: 'bool', value: true });
+  const variableContext = { classes: new Map(), localValues: variableLocalValues };
+  const variablePattern: VariablePattern = {
+    range: exampleRange,
+    type: 'VariablePattern',
+    name: 'var',
+  };
+  const variableExpression: SamlangExpression = EXPRESSION_VARIABLE({
+    range: exampleRange,
+    type: intType,
+    name: 'var',
+  });
+  const variableStatement: SamlangValStatement = {
+    pattern: variablePattern,
+    range: exampleRange,
+    typeAnnotation: intType,
+    assignedExpression: variableExpression,
+  };
+  const wildCardPattern: WildCardPattern = { type: 'WildCardPattern', range: exampleRange };
+  const wildCardStatement: SamlangValStatement = {
+    pattern: wildCardPattern,
+    range: exampleRange,
+    typeAnnotation: intType,
+    assignedExpression: intLiteralExpression,
+  };
+  const statementBlock: StatementBlock = {
+    range: exampleRange,
+    statements: [
+      tupleStatement,
+      tupleStatementNull,
+      objectStatement,
+      variableStatement,
+      wildCardStatement,
+    ],
+  };
+  const statementBlockWithExpression: StatementBlock = {
+    range: exampleRange,
+    statements: [],
+    expression: intLiteralExpression,
+  };
+  const statementBlockFail: StatementBlock = {
+    range: exampleRange,
+    statements: [objectStatementFail],
+  };
+  const statementBlockExpression = EXPRESSION_STATEMENT_BLOCK({
+    range: exampleRange,
+    type: intType,
+    block: statementBlock,
+  });
+  const statementBlockExpressionWithBlockExpression = EXPRESSION_STATEMENT_BLOCK({
+    range: exampleRange,
+    type: intType,
+    block: statementBlockWithExpression,
+  });
+  const statementBlockExpressionFail = EXPRESSION_STATEMENT_BLOCK({
+    range: exampleRange,
+    type: intType,
+    block: statementBlockFail,
+  });
+  expect(
```

Comment:
We need a nested statement block unit test to test that variables are indeed block scoped.

Reply:
ok I still need to do this but I'm assuming this means testing block statement with a nested block expression?

Reply:
yes